transcend/resource_data_silo.go

```diff
@@ -487,15 +492,28 @@ func resourceDataSilosUpdate(ctx context.Context, d *schema.ResourceData, m inte
 	var saasContext []byte
 	if d.Get("secret_context") != nil {
 		// Lookup the sombra URL to talk to
-		var sombraUrlQuery struct {
+		var queryBySombraId struct {
+			Sombras struct {
+				CustomerUrl  graphql.String `graphql:"customerUrl"`
+				HostedMethod graphql.String `graphql:"hostedMethod"`
+			} `graphql:"sombras(ids: [$sombra_id])"`
+		}
+		var queryPrimarySombra struct {
 			Organization struct {
 				Sombra struct {
 					CustomerUrl  graphql.String `graphql:"customerUrl"`
 					HostedMethod graphql.String `graphql:"hostedMethod"`
 				} `graphql:"sombra"`
 			} `graphql:"organization"`
 		}
-		err = client.graphql.Query(context.Background(), &sombraUrlQuery, map[string]interface{}{}, graphql.OperationName("SombraUrlQuery"))
+
+		var err error
+		if d.Get("sombra_id") != nil {
```

Comment:
Shouldn't this be?

```suggestion
		if d.Get("sombra_id") == nil {
```

Reply:
Thanks for catching this!! I fixed it and it should now represent the following logic: If there is no sombra_id included, we want to use the query grabbing the primary sombra. If there is an id, use the query that includes the filter for that id.